Iamsobanjaved/celery logs

iamsobanjaved opened the pull request:
PR created using tag v4.4.7. 

I have created a branch `awais786/v4.4.7` using `v4.4.7` tag and merge this PR in that.

https://github.com/edx/celery/blob/awais786/v4.4.7/celery/__init__.py#L23

iamsobanjaved committed:
Dump all registered tasks on UnregisteredTask exception
Added log about all apps name while autodiscove
Added logs and print on all steps
Added identifier in log statements

awais786 commented:
@jmbowman Please review and share your thoughts.

jmbowman commented on celery/fixups/django.py:
```diff
@@ -91,7 +94,14 @@ def now(self, utc=False):
         from django.apps import apps
-        return [config.name for config in apps.get_app_configs()]
+        configs = []
+        for config in apps.get_app_configs():
```
You might want to also log/print some output right before this, just to confirm we get this far in case the problem is within `get_app_configs()` itself.

jmbowman commented on celery/loaders/base.py:
```diff
@@ -255,21 +266,33 @@ def find_related_module(package, related_name):
         import_exc_name = getattr(e, 'name', module_name)
         if import_exc_name is not None and import_exc_name != module_name:
+            logger.exception('EDX-CELERY-ErrorInImport: Found some error in %s', package)
+            print('EDX-CELERY-ErrorInImport: Found some error in %s' % package)
```
Maybe log/print the exception itself also, in case something silently swallows it later?

jmbowman reviewed (approved):
Looking good, just a couple of additional suggestions.  When I do this kind of debugging, I usually add 5-10 print statements in places I think should be on the execution path that's hitting problems, and see which of them actually get run (and in what order) before something goes wrong.  Even if I guessed wrong about where the problem is, it helps me understand how execution flows through the code a little better so I can pick better locations in the code for the next iteration of debugging.

awais786 committed:
Dump all registered tasks on UnregisteredTask exception
Added log about all apps name while autodiscove
Added logs and print on all steps
Added identifier in log statements
